swift-sdk/Internal/APNSTypeChecker.swift

```diff
@@ -59,12 +59,39 @@ struct APNSTypeChecker: APNSTypeCheckerProtocol {
     }
     
     static func readMobileProvision(fromPath path: String) -> [AnyHashable: Any] {
-        guard let asciiString = try? String(contentsOfFile: path, encoding: .ascii) else {
+        guard let data = try? Data(contentsOf: URL(fileURLWithPath: path)) else {
             ITBError("Could not read file: \(path)")
             return [:]
         }
+
+        let encodings: [String.Encoding] = [
+            .ascii,
+            .utf8,
+            .utf16,
+            .utf16BigEndian,
+            .utf16LittleEndian,
+            .utf32,
+            .utf32BigEndian,
+            .utf32LittleEndian,
+            .isoLatin1,
+            .macOSRoman
+        ]
```

Comment:
Which one actually works? Do we need to try all?

Reply:
`.isoLatin1` worked for iOS 18. Ascii works with pre iOS 18 devices. 
You think I should narrow down to two that I know works instead of this?

Reply:
Did you need to update this with the appropriate encodings?

Reply:
@Ayyanchira I think we should at least put the verified working ones at the top, to reduce the risk of a situation where the data can be decoded with an encoding, but is then decoded incorrectly, and the string is mangled/can't be interpreted.

Reply:
Trying everything and continuing with the first thing that decodes it doesn't 100% guarantee that the decoded value would be valid.